### ISSUE TITLE
Line of Sight Simplification Improvements

### ISSUE THREAD

**Issue by anlumo** [moo-man/FVTT-DD-Import#60]
Hello,

I'd like to propose an improvement on the algorithm behind the fidelity slider. Right now, it appears that the algorithm removes points solely based on adjacent line length. This produces results like this:

![lowfidelity](https://user-images.githubusercontent.com/644763/105325569-0da0d700-5bcd-11eb-937b-1d1d45e9b8c9.png)

It would be much better if the code would check the distance of a line between the two adjacent points and the point itself and only remove the them under a certain threshold. That way the simplification wouldn't lose corner points and would be closer to the original shape. To illustrate, I'm talking about the yellow line here:

![bettersimplification](https://user-images.githubusercontent.com/644763/105326256-d67ef580-5bcd-11eb-9b2e-0bc450240967.png)

Basically, it's a linear approximation with minimized error.

**Comment by m42e** [comment]
I like the idea. The algorithm was really simple and meant for cave walls, as they contained way too much points. Do you really need the path simplification for the rounded corners? Are you getting too much walls if you do not activate it at all?
